packages/zoe/src/zoeService/zoeSeat.js

```diff
@@ -1,108 +1,293 @@
-      doExit(zoeSeatAdmin);
-    },
-    fail: reason => {
-      assert(
-        !hasExited(zoeSeatAdmin),
-        'Cannot fail seat. Seat has already exited',
-      );
-      updater.fail(reason);
-      doExit(zoeSeatAdmin);
-    },
-    getNotifier: () => Promise.resolve(notifier),
-  });
-
-  /** @type {UserSeat} */
-  const userSeat = Far('userSeat', {
-    getProposal: async () => proposal,
-    getPayouts: async () => payoutPromiseKit.promise,
-    getPayout: async keyword => {
-      assert(keyword, 'A keyword must be provided');
-      return E.get(payoutPromiseKit.promise)[keyword];
-    },
-    getOfferResult: async () => offerResult,
-    hasExited: async () => hasExited(zoeSeatAdmin),
-    tryExit: async () => E(exitObj).exit(),
-
-    getCurrentAllocationJig: async () => currentAllocation,
-    getAllocationNotifierJig: async () => notifier,
-    getFinalAllocation: () =>
-      E.when(payoutPromiseKit.promise, () => currentAllocation),
-
-    numWantsSatisfied: async () => {
-      return E.when(payoutPromiseKit.promise, () =>
-        satisfiesWant(proposal, currentAllocation),
-      );
+  // There is a race between resolveExitAndResult() and getOfferResult() that
+  // can be limited to when the adminFactory is paged in. If state.offerResult
+  // is defined, getOfferResult will return it. If it's not defined when
+  // getOfferResult is called, create a promiseKit, return the promise and store
+  // the kit here. When resolveExitAndResult() is called, it saves
+  // state.offerResult and resolves the promise if it exists, then removes the
+  // table entry.
+  const ephemeralOfferResultStore = new Map();
+
+  return vivifyFarClassKit(
+    baggage,
+    'ZoeSeatKit',
+    ZoeSeatIKit,
+    (
+      initialAllocation,
+      proposal,
+      instanceAdminHelper,
+      withdrawFacet,
+      exitObj = undefined,
+      // emptySeatKits start with offerResult validly undefined; others can set
+      // it to anything (including undefined) in resolveExitAndResult()
+      offerResultIsUndefined = false,
+    ) => {
+      const { publisher, subscriber } = makeDurablePublishKit();
+      return {
+        currentAllocation: initialAllocation,
+        proposal,
+        exitObj,
+        offerResult: undefined,
+        offerResultValid: offerResultIsUndefined,
+        instanceAdminHelper,
+        withdrawFacet,
+        publisher,
+        subscriber,
+        payouts: undefined,
+        exiting: false,
+      };
     },
-  });
+    {
+      zoeSeatAdmin: {
+        replaceAllocation(replacementAllocation) {
+          const { state } = this;
+          assertHasNotExited(
+            this,
+            'Cannot replace allocation. Seat has already exited',
+          );
+          harden(replacementAllocation);
+          // Merging happens in ZCF, so replacementAllocation can
+          // replace the old allocation entirely.
+
+          state.currentAllocation = replacementAllocation;
+        },
+        exit(reason) {
```

Comment:
```suggestion
        exit(completion) {
```
seems more consistent with usage elsewhere, and a good distinction to make.

Reply:
done